Apps/Src/Tasks.c

```diff
@@ -68,20 +68,24 @@ OS_SEM FaultState_Sem4;
 /**
  * Global Variables
  */
-
 // TODO: Put all global state variables here
+State RegenEnable = OFF;
+State CruiseControlEnable = OFF;
 
 // Needs to get initialized somewhere, not currently initialized
 fault_bitmap_t FaultBitmap;
 os_error_loc_t OSErrLocBitmap = OS_NONE_LOC;
 
-void assertOSError(uint16_t OS_err_loc, OS_ERR err){
-    if(err != OS_ERR_NONE){
+void assertOSError(uint16_t OS_err_loc, OS_ERR err)
+{
+    if (err != OS_ERR_NONE)
+    {
         FaultBitmap.Fault_OS = 1;
         OSErrLocBitmap |= OS_err_loc;
-        
+
         OSSemPost(&FaultState_Sem4, OS_OPT_POST_1, &err);
-        if(err != OS_ERR_NONE){
+        if (err != OS_ERR_NONE)
+        {
             EnterFaultState();
         }
```

Comment:
```suggestion
        EnterFaultState();
```
We probably want to enter a fault state if we reach this line, even if the semaphore is successfully posted